lib/mongodb.js

```diff
@@ -57,6 +57,11 @@ function generateMongoDBURL(options) {
   options.port = options.port || 27017;
   options.database = options.database || options.db || 'test';
   var username = options.username || options.user;
+  var portUrl = '';
+  // only include port if not using mongodb+srv
+  if(options.protocol !== 'mongodb+srv') {
```

Comment:
Nitpick: can we use a truthy assertion here?
Like 

```js
if(options.protocal === 'mongodb+srv') {
  options.port = '';
}
```

And then you don't have to make the code change on line 74 and 83.

Reply:
Not sure I follow, I need the change on 74 and 83 to remove the : as well as the port number. Pretty sure the url will not be valid with : in it.